src/collection.js

```diff
@@ -460,6 +460,8 @@ export default class Collection {
    * @return {Promise}
    */
   importChanges(syncResultObject, changeObject) {
+    // XXX: Ensure all imports are done within a single transaction
+    // Remove this and replace by batch(changes.map(change => batch[action](change)))
     return Promise.all(changeObject.changes.map(change => {
       return this._importChange(change);
     }))
```

Comment:
It would be nice if `batch(changes.map(change => batch[action](change)))` could return a promise for a results object like the one `Promise.all` returns. Alternatively, change the code in line https://github.com/Kinto/kinto.js/pull/180/files#diff-d2dbe7779d59d4dd594e92248caba0a5R469 to match the `{operations, errors}` syntax.